File: swift/Sources/Allmark/Utils/GetEndOfLine.swift
import Foundation

func getEndOfLine(state: inout BlockParserState) -> Int {
	var endOfLine = state.i
	while endOfLine < state.src.count {
		let char = state.src[endOfLine]
		if char == "\n" || char == "\r\n" || char == "\r" {
			endOfLine += 1
			state.lineStart = endOfLine
			break
		}
		endOfLine += 1
	}
	return endOfLine
}

File: web/src/utils/getEndOfLine.ts
import type BlockParserState from "../types/BlockParserState";
import { CARRIAGE_RETURN_CODE, NEW_LINE_CODE } from "./charCodes";

export default function getEndOfLine(state: BlockParserState): number {
	let endOfLine = state.i;
	for (; endOfLine < state.src.length; endOfLine++) {
		let code = state.src.charCodeAt(endOfLine);
		if (code === NEW_LINE_CODE) {
			endOfLine++;
			state.lineStart = endOfLine;
			break;
		} else if (code === CARRIAGE_RETURN_CODE) {
			endOfLine++;
			if (state.src.charCodeAt(endOfLine) === NEW_LINE_CODE) {
				endOfLine++;
			}
			state.lineStart = endOfLine;
			break;
		}
	}
	return endOfLine;
}
